### DISCUSSION TITLE
Issues bench testing fail safe and RTH

### DISCUSSION THREAD

**Discussion by mackboston** [iNavFlight/inav#9081] [Q&A] [answered]
When I test the RTH on my desk nothing happens. I believe when I used to test it on the desk it would start to bank the plane and speed up when I hit RTH. Any idea what’s going on? To test I’m arming the plane putting a tiny bit of throttle and then flipping the RTH button. It shows as activated in the mode tab.

In addition when I try to test fail safe by putting the motor to 1/4 power walking 50 meters away and turning off my transmitter it acts strange. The plane banks and motor slows to almost stopped. The plane however is set to climb and increase speed. I can’t figure out what’s going on with these tests. Are they no longer good bench tests for some reason?

Matek h743 wlite. Tx16s mark2 with ranger elrs. Matek ELRS-R24-D. Inav 6.1.0. Elrs 3.2.0

**Comment by Jetrell** [comment, marked as answer]
I assume you're aware RTH can't be tested on the bench, when your sitting still.. The controller requires a minimum velocity for GPS position reference.
When you have a GPS fix. You can dummy throw the plane, so it temporarily reaches that required speed, Then you can walk around with the plane, to perform a ground simulated RTH test... Provide you walk with, what the navigation system is trying to make the plane do.. Other wise the controller will see conflicting data, with the way the improved AHRS calculations work.

**Reply by mackboston** [reply]
Ok thanks. I didn’t know that. I guess I will just test it in the air. I assume it’s the same deal for failsafe mode?

**Reply by Jetrell** [reply]
Yes. The same for Failsafe mode.
